### ISSUE TITLE
[FEATURE]: update-remote command should fetch sources in parallel

### ISSUE THREAD

**Issue by hnicke** [garden-io/garden#3169]
### Background / Motivation

When running `garden update-remote sources`, garden updates the remotes sequentially.
This takes quite a while.

### What should the user be able to do?

`garden update-remote sources` should update the remotes in parallel, decreasing the wait time for the user.

### Why do they want to do this? What problem does it solve?

As a user, it takes way to long to wait for the `update-remote` command to finish.
As a workaround we might stop using `update-remote` command and instead use `garden link` for all sources all the time - and update the locally checked out remote sources with a third party helper / script (in parallel).

### How important is this feature for you/your team?

It’s a nice to have, but nice things are nice 🙂

**Comment by vvagaytsev** [comment]
Good point @hnicke, thanks for reporting this! We'll take a look soon.
